### PR TITLE
Override operator== for RealmObject

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

File: lib/src/realm_object.dart [hunk 1]
```diff
@@ -206,6 +206,16 @@ class RealmObject {
...
   }
+
+  @override
+  // ignore: hash_and_equals
+  bool operator ==(Object other) {
+    final o = other;
+    if (identical(this, o)) return true;
+    if (o is! RealmObject) return false;
+    if (!isManaged || !o.isManaged) return false;
+    return realmCore.equals(handle, o.handle);
```
Comment (blagoev) [inline lib/src/realm_object.dart:217]: I would suggest we follow the other places and keep passing realm objects to realm_core and get the handles from inside the realm_core implementation. I will put suggestions on the relevant places

File: lib/src/native/realm_core.dart [hunk 2]
```diff
@@ -442,6 +442,10 @@ class _RealmCore {
   void listClear(RealmList list) {
     _realmLib.invokeGetBool(() => _realmLib.realm_list_clear(list.handle._pointer));
   }
+
+  bool equals<T extends NativeType>(Handle<T> x, Handle<T> y) {
```
Comment (blagoev) [inline lib/src/native/realm_core.dart:446]:
```suggestion
  bool equals(RealmObject first, RealmObject second) {
```

File: lib/src/realm_object.dart (hunk 1 above)
Comment (blagoev) [inline lib/src/realm_object.dart:217]:
```suggestion
    return realmCore.equals(this, other);
```

File: lib/src/realm_object.dart [hunk 3]
```diff
@@ -206,6 +206,16 @@ class RealmObject {
     RealmAccessor.setDefaults<T>(values);
     return true;
   }
+
+  @override
+  // ignore: hash_and_equals
+  bool operator ==(Object other) {
+    final o = other;
```
Comment (blagoev) [inline lib/src/realm_object.dart:213]: perhaps we don't need an additional local variable.

File: test/realm_test.dart [hunk 4]
```diff
@@ -1073,5 +1069,29 @@ Future<void> main([List<String>? args]) async {
...
+        realm..add(person)..add(dog);
+      });
+
+      expect(person, person);
+      expect(person, isNot(1)); 
+      expect(person, isNot(dog));
+
+      final read = realm.query<Person>("name == 'Kasper'");
+
+      expect(read, [person]);
```
Comment (blagoev) [inline test/realm_test.dart:1094]: Minor: Not sure what the error message when comparing collections, I would have written `expect(read[0], person);` Hope it's readable.

**Review by blagoev — CHANGES_REQUESTED** [review]
I suggest some small changes

File: test/realm_test.dart (hunk 4 above)
Comment (blagoev) [reply test/realm_test.dart:1094]: consider ignoring this if you think its alright

File: lib/src/realm_object.dart (hunk 3 above)
Comment (nielsenko) [reply lib/src/realm_object.dart:213]: Fixed

File: lib/src/realm_object.dart (hunk 1 above)
Comment (nielsenko) [reply lib/src/realm_object.dart:217]: Fixed

File: lib/src/native/realm_core.dart (hunk 2 above)
Comment (nielsenko) [reply lib/src/native/realm_core.dart:446]: Fixed

File: lib/src/realm_object.dart (hunk 1 above)
Comment (nielsenko) [reply lib/src/realm_object.dart]: Fixed

File: test/realm_test.dart (hunk 4 above)
Comment (nielsenko) [reply test/realm_test.dart:1154]: I think it is a bit safer like this. If read is empty, read[0] throws before expect gets to run. Also, the output is quite nice, but we should consider generating a toString method for realm objects.

Comment (nielsenko) [reply test/realm_test.dart:1154]: Also `expect(read[0], person)` don't check if `read.length > 1`

File: lib/src/native/realm_core.dart [hunk 5]
```diff
@@ -442,6 +442,10 @@ class _RealmCore {
   void listClear(RealmList list) {
     _realmLib.invokeGetBool(() => _realmLib.realm_list_clear(list.handle._pointer));
   }
+
+  bool equals<T extends NativeType>(RealmObject first, RealmObject second) {
```
Comment (blagoev) [inline lib/src/native/realm_core.dart:446]:
```suggestion
  bool equals(RealmObject first, RealmObject second) {
```

File: test/realm_test.dart (hunk 4 above)
Comment (blagoev) [reply test/realm_test.dart:1154]: sure, scratch my comment

File: lib/src/native/realm_core.dart (hunk 2 above)
Comment (blagoev) [reply lib/src/native/realm_core.dart]: I think we don't need the type constraint on this method.  Added a suggestion

File: test/realm_test.dart (hunk 4 above)
Comment (blagoev) [reply test/realm_test.dart:1154]: scratch my comment

**Review by blagoev — APPROVED** [review]

**Pull request merged**